Backend/sub_crates/language/src/tests/get.rs

```diff
@@ -34,4 +34,165 @@ mod tests {
     dictionary.register("Test", Language::Japanese, "test");
     assert_eq!(dictionary.get("Test", Language::Japanese), "test")
   }
+  #[test]
+  #[should_panic(expected = "Value is empty for key Test and language 0")]
+  fn has_empty_translation() {
+    let dictionary = Dictionary::default();
+    dictionary.register("Test", Language::English, "");
+    dictionary.get("Test", Language::English);
+  }
+
+
+  /*
+  Tests derived by input space partitioning with the following characteristics:
+  empty/nonempty key
+  key pure ASCII or with at least one non-ASCII character
+  0, 1, >1 translations
+  translation for requested is available or not
+  translation is valid or not (empty string or nonempty string)
+   */
+  #[test]
+  #[should_panic(expected = "Key  is not registered")]
+  fn keyEmpty_0Translations() {
+    let dictionary = Dictionary::default();
+    dictionary.get("", Language::English);
+  }
+
+  #[test]
+  #[should_panic(expected = "Key äöüß is not registered")]
+  fn keyNonEmpty_nonAscii_0Translations() {
+    let dictionary = Dictionary::default();
+    dictionary.get("äöüß", Language::English);
+  }
+
+  #[test]
+  #[should_panic(expected = "Key Test is not registered")]
+  fn keyNonEmpty_ascii_0Translations() {
+    let dictionary = Dictionary::default();
+    dictionary.get("Test", Language::English);
+  }
+
+  #[test]
+  fn keyEmpty_1Translations_available_valid() {
+    let dictionary = Dictionary::default();
+    dictionary.register("", Language::English, "test");
+    assert_eq!(dictionary.get("", Language::English), "test");
+  }
+
+  #[test]
+  // #[should_panic(expected = "Value is empty for key Test and language 0")]
```

Comment:
Dead code?

Reply:
That's part of the WIP. All those expectation pairs are for the spec "as is" vs "as I would expect"